hnn_core/gui/gui.py

```diff
@@ -39,232 +58,361 @@
-                step_f = 1.0
-                freqs = np.arange(min_f, max_f, step_f)
-                n_cycles = freqs / 8.
-                fig, ax = plt.subplots()
-                dpls_copied[0].plot_tfr_morlet(
-                    freqs,
-                    n_cycles=n_cycles,
-                    colormap=analysis_config['spectrogram_cm'],
-                    ax=ax)
-            else:
-                print("No dipole data")
-        elif plot_type == 'network':
-            if net_copied:
-                fig = plt.figure()
-                ax = fig.add_subplot(111, projection='3d')
-                net_copied.plot_cells(ax=ax)
-            else:
-                print("No network data")
-
-
-def _gen_plot_callback(simulation_data, analysis_config, plot_key):
-    def callback(_):
-        _update_plot_window(simulation_data, analysis_config, plot_key)
-    return callback
-
-
-def _init_viz_layout(simulation_data, plot_options, previous_plot_types,
-                     analysis_config, style, layout_option, init=False):
-    plot_outputs = analysis_config['plot_outputs']
-    plot_dropdowns = analysis_config['plot_dropdowns']
-    plot_sim_selections = analysis_config['plot_sim_selections']
-
-    sim_names = list(simulation_data.keys())
-    assert analysis_config['current_sim_name'] in sim_names
-
-    if layout_option == "L-R":
-        # TODO fine tune the style so that there's no more overflow.
-        grid_style = Layout(width=f"{int(style.width[:-2])/2-10}px",
-                            # height=style.height,
-                            height=f"{int(style.height[:-2])-80}px",
-                            border=style.border)
-        grid = GridspecLayout(1, 2, height=style.height)
-        plot_keys = ["Left", "Right"]
-
-    elif layout_option == "U-D":
-        grid_style = Layout(width=f"{int(style.width[:-2])/2-10}px",
-                            height=f"{float(style.height[:-2])/2-10}px",
-                            border=style.border,)
-        grid = GridspecLayout(2, 1, height=style.height)
-        plot_keys = ["Up", "Down"]
-
-    for idx, plot_key in enumerate(plot_keys):
-        plot_outputs[plot_key] = Output(layout=grid_style)
-
-        plot_dropdowns[plot_key] = Dropdown(
-            options=plot_options,
-            value=previous_plot_types[plot_key]
-            if plot_key in previous_plot_types else plot_options[idx],
-            description='Plot:',
-            disabled=False,
-            layout=Layout()
-        )
-        plot_dropdowns[plot_key].observe(
-            _gen_plot_callback(simulation_data, analysis_config, plot_key),
-            'value',
-        )
+        w_dipole_smooth = FloatText(value=plot_spec['config']['dipole_smooth'],
+                                    description='Dipole Smooth Window (ms):',
+                                    disabled=False,
+                                    style=analysis_style)
+
+        def _handle_dsm_change(b):
```

Comment:
spell out `dsc` and `dsm` ... I just can't remember what they stand for. Maybe you can say `_onchange_smooth_window` or something of that sort